metadata-io/src/main/java/com/linkedin/metadata/entity/cassandra/CassandraAspectDao.java

```diff
@@ -148,10 +150,13 @@ private Map<String, Long> getMaxVersions(@Nonnull final String urn, @Nonnull fin
   }
 
   @Override
-  public void saveAspect(@Nonnull EntityAspect aspect, final boolean insert) {
+  public void saveAspect(@Nonnull EntityAspect aspect, final boolean insert, @Nullable ExtraIngestParams extraIngestParams) {
     validateConnection();
-    SimpleStatement statement = generateSaveStatement(aspect, insert);
-    _cqlSession.execute(statement);
+    SimpleStatement statement = generateSaveStatement(aspect, insert, extraIngestParams);
+    ResultSet resultSet = _cqlSession.execute(statement);
+    if (!resultSet.wasApplied()) {
+      throw new PreconditionFailedException("Condition for Conditional Update is not met");
```

Comment:
Can this be changed in line with the suggestions above to add urn+aspect details and some expectations to the error message?